edge-apis/credentials.go

```diff
@@ -71,34 +78,42 @@ func getClientAuthInfoOp(credentials Credentials, client *http.Client) func(*run
 
 // BaseCredentials is a shared struct of information all Credentials implementations require.
 type BaseCredentials struct {
-	//ConfigTypes is used to set the configuration types for services during authentication
+	// ConfigTypes is used to set the configuration types for services during authentication
 	ConfigTypes []string
 
-	//EnvInfo is provided during authentication to set environmental information about the client.
+	// Headers is a map of strings to string arrays of headers to send with auth requests.
+	Headers *http.Header
+
+	// EnvInfo is provided during authentication to set environmental information about the client.
 	EnvInfo *rest_model.EnvInfo
 
-	//SdkInfo is provided during authentication to set SDK information about the client.
+	// SdkInfo is provided during authentication to set SDK information about the client.
 	SdkInfo *rest_model.SdkInfo
 
-	//CaPool will override the client's default certificate pool if set to a non-nil value.
+	// CaPool will override the client's default certificate pool if set to a non-nil value.
 	CaPool *x509.CertPool
 }
 
+// NewCredentials creates a new BaseCredentials instance.
+func NewCredentials() *BaseCredentials {
```

Comment:
This should not exist.

Reply:
This is how I was using it: https://github.com/openziti-test-kitchen/zssh/blob/37c5561e1747ea312584487ca6ede3502702713c/zsshlib/ssh.go#L329

The problem is that, short of that, one needs to create another Credentials type that doesn't match the intent, making it less expressive... for instance, I could change the accessor to perform the following:

```
conf := getConfig(f.ZConfig)
conf.Credentials = edge_apis.NewJwtCredentials("")
conf.Credentials.AddHeader("Authorization", "Bearer "+token)
```
which would work, but gives the misconception that the JwtCredential type is actually being leveraged or is necessary.